backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/AggregationDAO.java

```diff
@@ -129,7 +127,39 @@ public int findC_Invoice_Candidate_HeaderAggregationKey_ID(I_C_Invoice_Candidate
 		headerAggregationKeyRecord.setC_BPartner_ID(bpartnerId);
 		headerAggregationKeyRecord.setIsSOTrx(ic.isSOTrx());
 		headerAggregationKeyRecord.setIsActive(true);
-		InterfaceWrapperHelper.save(headerAggregationKeyRecord);
-		return headerAggregationKeyRecord.getC_Invoice_Candidate_HeaderAggregation_ID();
+		return saveHeaderAggregationKeyWithLookupOnFail(headerAggregationKeyRecord, ic, headerAggregationKeyCalc);
+	}
+
+	private int saveHeaderAggregationKeyWithLookupOnFail(
+			@NonNull final I_C_Invoice_Candidate_HeaderAggregation headerAggregationKeyRecord,
+			@NonNull final I_C_Invoice_Candidate ic,
+			@NonNull final String headerAggregationKeyCalc)
+	{
+		try
+		{
+			InterfaceWrapperHelper.save(headerAggregationKeyRecord);
+			return headerAggregationKeyRecord.getC_Invoice_Candidate_HeaderAggregation_ID();
+		}
+		catch (final DBUniqueConstraintException e)
+		{
+			final int lookupHeaderAggregationKeyId = lookupHeaderAggregationKeyId(ic, headerAggregationKeyCalc);
+
+			if (lookupHeaderAggregationKeyId <= 0)
+			{
+				throw e;
+			}
+			return lookupHeaderAggregationKeyId;
+		}
+	}
+
+	private int lookupHeaderAggregationKeyId(
+			@NonNull final I_C_Invoice_Candidate ic,
+			@NonNull final String headerAggregationKeyCalc)
+	{
+		return Services.get(IQueryBL.class)
+				.createQueryBuilder(I_C_Invoice_Candidate_HeaderAggregation.class, ic)
```

Comment:
I know it was already there, but pls extract `Services.get(IQueryBL.class)` in an instance field.